### PR TITLE
FISH-6971 Instrument JAX-RS endpoints in OTEL-enabled application

### PR DESCRIPTION
**Pull request by arieki**
## Description

Enable OpenTelemetry on JAX-RS server-side

### Blockers

N/A

### New tests

N/A

### Testing Performed

Build and run Payara server, execute MP OpenTelemetry TCK and MP Opentracing TCK

### Testing Environment

Windows 11, OpenJDK11, Maven 3.8.4

## Notes for Reviewers

To be able to execute MP OpenTelemetry TCK there is one adjustment that needs to do which is enabling `JerseyOpenTelemetryAutoDiscoverable` to be automatically discovered by JAX-RS runtime. This is necessary because when I enable OpenTelemetry by default, Payara server loose compatibility with opentracing spec

### CODE REVIEW

**Comment by arieki** [comment]
jenkins test please

**Review by pdudits — CHANGES_REQUESTED** [review]
I'll do more throughout review when I check out the branch.

It is heading in right direction, we might need to make configuration more consistent and don't make TCK-specific workarounds. If anything, those should be configurable in TCK runner.

File: appserver/payara-appserver-modules/jaxrs-client-tracing/pom.xml [hunk 1]
```diff
@@ -71,6 +71,15 @@
...
         </dependency>
+        <dependency>
+            <groupId>io.opentelemetry</groupId>
+            <artifactId>opentelemetry-api</artifactId>
+            <scope>provided</scope>
+        </dependency>
+<!--        <dependency>-->
+<!--            <groupId>io.opentelemetry</groupId>-->
+<!--            <artifactId>opentelemetry-semconv</artifactId>-->
+<!--        </dependency>-->
```
Comment (pdudits) [inline appserver/payara-appserver-modules/jaxrs-client-tracing/pom.xml:82]: Remove it in next PR, please

**Review by pdudits — APPROVED** [review]

**Comment by arieki** [comment]
jenkins test please

**Pull request merged**